intermediate_source/inductor_debug_cpu.py

```diff
@@ -0,0 +1,600 @@
+#
+# (1) Regard to ``mkl::_mkl_linear``: You may notice the number of calls to this kernel is 362, which is exactly the same as ``aten::linear`` in the eager model profiling table.
+# The CPU total of ``aten::linear`` is 376.888ms, at the mean time it is 231.573ms for ``mkl::_mkl_linear``. This suggests inductor model speed up ~1.63x for the "linear" part.
+#
+# (2) Regarding non-linear part: The end-to-end latency for the eager/inductor model is 802/339ms. The speed up for the non-linear part is ~3.94x.
+# Let's read the generated code to understand how the inductor achieves this impressive optimization. You are able to find the generated code by 
+# searching ``cpp_fused__mkl_linear_add_mul_relu_151`` in ``output_code.py``
+# 
+
+
+cpp_fused__mkl_linear_add_mul_relu_151 = async_compile.cpp('''
+#include <ATen/record_function.h>
+#include "/tmp/torchinductor_root/lr/clrlgu27q4ggd472umdzwsu6qcpqxcuusjxqvx2hwitjbujiiz7z.h"
+extern "C" void kernel(float* in_out_ptr0,
+                       const float* in_ptr0,
+                       const float* in_ptr1,
+                       const float* in_ptr2,
+                       const float* in_ptr3)
+{
+    RECORD_FUNCTION("graph_0_cpp_fused__mkl_linear_add_mul_relu_151", c10::ArrayRef<c10::IValue>({}));
+    #pragma omp parallel num_threads(32)
+    {
+        {
+            #pragma omp for 
+            for(long i0=static_cast<long>(0L); i0<static_cast<long>(16384L); i0+=static_cast<long>(1L))
+            {
+                for(long i1=static_cast<long>(0L); i1<static_cast<long>(512L); i1+=static_cast<long>(8L))
+                {
+                    auto tmp0 = at::vec::Vectorized<float>::loadu(in_ptr0 + static_cast<long>(i1 + (512L*i0)));
+                    auto tmp1 = at::vec::Vectorized<float>::loadu(in_ptr1 + static_cast<long>(i1));
+                    auto tmp3 = at::vec::Vectorized<float>::loadu(in_out_ptr0 + static_cast<long>(i1 + (512L*i0)));
+                    auto tmp5 = at::vec::Vectorized<float>::loadu(in_ptr2 + static_cast<long>(i1));
+                    auto tmp7 = at::vec::Vectorized<float>::loadu(in_ptr3 + static_cast<long>(i1));
+                    auto tmp2 = tmp0 + tmp1;
+                    auto tmp4 = tmp2 + tmp3;
+                    auto tmp6 = tmp4 * tmp5;
+                    auto tmp8 = tmp6 + tmp7;
+                    tmp8.store(in_out_ptr0 + static_cast<long>(i1 + (512L*i0)));
+                }
+            }
+        }
+    }
+}
+''')
+
+######################################################################
+# From the generated code above, we can see this kernel has done a typical `Loop Fusion <https://en.wikipedia.org/wiki/Loop_fission_and_fusion>`_ on [add, add, mul, add].
+# We can infer the sizes and stride of the inputs and further bench this [add, add, mul, add] pattern.
+
+import torch
+def func(x0, x1, x3, x5, x7):
+    x2 = x0 + x1
+    x4 = x2 + x3
+    x6 = x4 * x5
+    x8 = x6 + x7
+    x3 = x8
+    return x3
```

Comment:
It is a bit awkward to manually construct a case here. Can we just compare the perf with the corresponding ops from the eager-mode model? There are "add" and "mul" showing up the eager model profiling too.

Reply:
I think this part is a good case to help users have an intuitive feeling about loop fusion.
I renamed the args `x0`, `x0`, ... , maybe it is better now.